app/controllers/tcsp/ComplexCorpStructureCreationController.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.tcsp
+
+import connectors.DataCacheConnector
+import controllers.BaseController
+import forms.{EmptyForm, Form2, InvalidForm, ValidForm}
+import javax.inject.Inject
+import models.tcsp._
+import uk.gov.hmrc.play.frontend.auth.connectors.AuthConnector
+import views.html.tcsp._
+
+import scala.concurrent.Future
+
+class ComplexCorpStructureCreationController @Inject()(
+                                                 val authConnector: AuthConnector,
+                                                 val dataCacheConnector: DataCacheConnector
+                                               ) extends BaseController {
+
+  val NAME = "complexCorpStructureCreation"
+  implicit val boolWrite = utils.BooleanFormReadWrite.formWrites(NAME)
+  implicit val boolRead = utils.BooleanFormReadWrite.formRule(NAME, "error.required.tcsp.complex.corporate.structures")
+
```

Comment:
I know it was already reviewed, just wondering why above is here?

Reply:
I also wanted to ask about this. The majority if not all of our reads/writes are in models within our service. Seems odd to branch away from that consistency we currently have.